### PR TITLE
Topic Recognition Final Report Joseph Savage s4647117

### PR DESCRIPTION
**Pull request by JosephSav**
Final Report submission.
3DUnet segmentation of the prostate dataset.
by Joseph Savage s4647117

### CODE REVIEW

**Comment by celinsze** [comment]
**This is an initial inspection, no action is required at this point**
Recognition problem: total 13 Marks
Problem solving: (4)
•	Algorithm was able to solve the problem with average dice score above 0.87 for each class. 
•	Missing plots for losses and training metrics.
Algorithm implementation: (2)
•	Algorithm implemented in PyTorch for the UNet3D model and functions.
•	Missing predict.py file for testing.
Design: Good coding design (1)
Commenting: Sufficient comments throughout the code (1)
Difficulty: 3D UNet – Normal (5)

**Comment by JosephSav** [comment]
Hi,
Just wanted to comment that I was specifically told by a tutor on Ed when I asked that I did not require the predict.py file for this project.

**Comment by celinsze** [comment]
Hi,
According to the task sheet provided, the predict.py is required under section 4 for model testing.

**Comment by JosephSav** [comment]
https://edstem.org/au/courses/18266/discussion/2304251
Zhixuan Li said "that's not required for each task tho" when I asked what to do in predict.py.
And after I told him I was doing 3DUNet he responded:
"The task sheet only asks for Dice similarity, which can be calculated in the training loop. So in your case I think that's all you expected to do"

**Comment by Uto0op** [comment]
Yes, but still I think you gave me wrong. Ideally `predict.py` should be used to load the saved model and compute something (e.g. metrics, reconstructed/generated images) for testing purpose. In your case, since you already outputted metrics during training, I wouldn't take marks off for not doing it again in `predict.py` --- that's what I meant by "all you expected to do".
By responding so, I did not "specifically" suggest no `predict.py` required, and you should always refer to the task sheet. I apologize if my response was confusing.

**Comment by JosephSav** [comment]
It's frustrating to lose marks on something like this, especially after the effort I put into it.
May I also ask what loss/training metric plots I am missing?

**Comment by Uto0op** [comment]
I understand, that's why we have this feedback mechanism, to get lost marks back by incorporation :)
For the `predict.py`, you can simply create one, load your saved model and do something in it; For the losses/metrics, you might need to start training again if you didn't save them.

**Comment by JosephSav** [comment]
Oh I didn't realise this was feedback I could act on.
Which loss/metrics am I missing at the moment though?

**Comment by Uto0op** [comment]
I'll leave the question to your marker @celinsze :)

**Comment by celinsze** [comment]
Hi Joseph, that was referring to the plot for losses during the training epochs and images for example outputs. Those are mentioned in your document but no images were displayed and they were not included in the folder.

**Comment by JosephSav** [comment]
That's strange I can see them fine on mine.
I've uploaded all my images to a separate public repo and just linking to hat in HTML.
Can you see any images/gifs at all in my report?

**Comment by celinsze** [comment]
From my side the only images visible was the two input images and overlay gif. I can see the html links from the ReadMe file but they were not showing on my computer. It might be better to add the image files into the folder to ensure they could be viewed.

**Comment by JosephSav** [comment]
Hi,
I changed all the images/gifs to be relative links and the images are uploaded in the repo.
Could you let me know if these are now visible?

**Comment by celinsze** [comment]
Yes I can see them now

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Normal. UNet3D (Normal Difficulty)</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. </td><td>-1</td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time submission.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is missing. </td><td>-2</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-9</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by JosephSav** [comment]
I do have header blocks on all my code files.
What was I meant to put into the request description to not lose a mark on that?

**Comment by shakes76** [comment]
Segmentations not one hot encoded -2 additionally

**Comment by shakes76** [comment]
Header blocks present and description OK +2
